mypy/fastparse.py

```diff
@@ -271,7 +271,20 @@ def visit(self, node: Optional[AST]) -> Any:
             method = 'visit_' + node.__class__.__name__
             visitor = getattr(self, method)
             self.visitor_cache[typeobj] = visitor
-        return visitor(node)
+        result = visitor(node)
+        if (3, 8) < sys.version_info and isinstance(node, ast3.expr):
+            self.scope_ignores(node)
+        return result
+
+    def scope_ignores(self, node: ast3.expr) -> None:
+        end_lineno = getattr(node, "end_lineno", None)
+        if end_lineno is None:
+            return
+        node_lines = range(node.lineno, end_lineno + 1)
+        for line in node_lines:
+            if line in self.type_ignores.values():
+                self.type_ignores = {**dict.fromkeys(node_lines, line), **self.type_ignores}
```

Comment:
@msullivan does mypyc support this?

Reply:
This has been refactored not to use unpacking, either way!